### PR TITLE
Preserve data

### PR DESCRIPTION
**Pull request by Omwomo**
- Serialized data for people, books, and rentals.
- Saved data for people, books, and rentals into their respective json files.
- Deserialized data for people, books, and rentals when loading.
- Handled errors when loading the data.

### CODE REVIEW

File: storage/books.json [hunk 1]
```diff
@@ -0,0 +1 @@
..."{\"title\":\"mastery\",\"author\":\"robert\"}","{\"title\":\"a\",\"author\":\"a\"}","{\"title\":\"Atomic Habbits\",\"author\":\"Eugene\"}","{\"title\":\"Mastery\",\"author\":\"Robert\"}","{\"title\":\"Mastery\",\"author\":\"Robert\"}","{\"title\":\"a\",\"author\":\"a\"}","{\"title\":\"a\",\"author\":\"a\"}","{\"title\":\"a\",\"author\":\"a\"}","{\"title\":\"a\",\"author\":\"a\"}","{\"title\":\"a\",\"author\":\"a\"}","{\"title\":\"a\",\"author\":\"a\"}","{\"title\":\"a\",\"author\":\"a\"}","{\"title\":\"a\",\"author\":\"a\"}","{\"title\":\"a\",\"author\":\"a\"}","{\"title\":\"z\",\"author\":\"z\"}","{\"title\":\"z\",\"author\":\"z\"}","{\"title\":\"a\",\"author\":\"a\"}","{\"title\":\"z\",\"author\":\"z\"}","{\"title\":\"x\",\"author\":\"x\"}","{\"title\":\"z\",\"author\":\"z\"}","{\"title\":\"x\",\"author\":\"x\"}","{\"title\":\"c\",\"author\":\"c\"}","{\"title\":\"v\",\"author\":\"v\"}","{\"title\":\"z\",\"author\":\"z\"}","{\"title\":\"x\",\"author\":\"x\"}","{\"title\":\"z\",\"author\":\"z\"}","{\"title\":\"x\",\"author\":\"x\"}","{\"title\":\"x\",\"author\":\"x\"}","{\"title\":\"a\",\"author\":\"a\"}","{\"title\":\"ty\",\"author\":\"ty\"}","{\"title\":\"d\",\"author\":\"d\"}"]
```
Comment (skyv26) [inline storage/books.json]: - [ ] Please remove the `redundancy` from your code, I tried to run the application and went with option 1 to `list` down all the books, but I was not able to get the book information that you have stored, instead I was getting the `blank` author, title data. 

Screenshot from 2023-11-01 02-52-06

- [ ] The second problem, that I noted down is that you haven't added the `try` catch in your code to handle the exceptions and errors in your code, just imagine, by mistake user deleted the `books.json` file from your folder and if user run the application, it will give an error regarding, file not found. I would like to suggest you find a way, if the file doesn't exist then on the spot the file is created by your application and serves its purpose.

- [ ] The Third problem that I faced was, if a person manually adds the empty array inside this file `so` that the application should work smoothly, I noted that your data is not converted to `JSON` format properly. Because the data you are preserving is not in the correct format. maybe this is the reason, while listing the book, I was not able to get the preserve data on the console display. Kindly fix the things test your application thorougly with all edge cases.

Screenshot from 2023-11-01 02-56-08

File: storage/people.json [hunk 2]
```diff
@@ -0,0 +1 @@
...,\"name\":\"a\",\"age\":3}","{\"id\":547,\"name\":\"omwom\",\"age\":44}","{\"id\":26,\"name\":\"a\",\"age\":3}","{\"id\":598,\"name\":\"Ofafa\",\"age\":26}","{\"id\":403,\"name\":\"Ofafa\",\"age\":26}","{\"id\":613,\"name\":\"a\",\"age\":3}","{\"id\":248,\"name\":\"a\",\"age\":1}","{\"id\":882,\"name\":\"a\",\"age\":0}","{\"id\":607,\"name\":\"a\",\"age\":1}","{\"id\":268,\"name\":\"q\",\"age\":1}","{\"id\":944,\"name\":\"q\",\"age\":1}","{\"id\":64,\"name\":\"q\",\"age\":1}","{\"id\":862,\"name\":\"q\",\"age\":1}","{\"id\":230,\"name\":\"q\",\"age\":1}","{\"id\":580,\"name\":\"q\",\"age\":1}","{\"id\":809,\"name\":\"w\",\"age\":2}","{\"id\":19,\"name\":\"q\",\"age\":1}","{\"id\":970,\"name\":\"q\",\"age\":1}","{\"id\":249,\"name\":\"q\",\"age\":1}","{\"id\":792,\"name\":\"w\",\"age\":2}","{\"id\":434,\"name\":\"q\",\"age\":2}","{\"id\":484,\"name\":\"c\",\"age\":0}","{\"id\":604,\"name\":\"q\",\"age\":2}","{\"id\":393,\"name\":\"q\",\"age\":1}","{\"id\":385,\"name\":\"w\",\"age\":2}","{\"id\":872,\"name\":\"q\",\"age\":1}","{\"id\":700,\"name\":\"w\",\"age\":2}","{\"id\":533,\"name\":\"w\",\"age\":2}","{\"id\":366,\"name\":\"q\",\"age\":1}","{\"id\":966,\"name\":\"s\",\"age\":0}"]
```
Comment (skyv26) [inline storage/people.json]: - [ ] As I mentioned in my above `inline-comment` the same is happening here, once you fix the `books` JSON file formatting and save the data appropriately then take a look into this file too, and implement the things that I mentioned earlier.

**Review by skyv26 — CHANGES_REQUESTED** [review]
## Hi @Omwomo, @BiancaMe and @axcell-0, 

Good job so far 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## STATUS: REQUIRE CHANGES 🔴 



## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Professional `README` template ✅ 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

#### Cheers, and Happy coding!👏👏👏

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mukhammadrizooff — APPROVED** [review]
Hi @TEAM, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

 

## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
